File: app/src/main/java/io/datalive/myreferrer/InstallActivity.java
package io.datalive.myreferrer;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class InstallActivity extends AppCompatActivity {

    private TextView txtReferrer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_install);

        txtReferrer = (TextView) findViewById(R.id.txtReferrer);

        Intent intent = getIntent();
        Uri uri = intent.getData();
        if(uri != null) {
            String v1 = uri.getQueryParameter("k1");
            String v2 = uri.getQueryParameter("k2");

            txtReferrer.setText("V1 : " + v1 + ", V2 : " + v2);
        }
    }
}

File: app/src/main/java/io/datalive/myreferrer/InstallActivity.kt
package io.datalive.myreferrer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_install.*

class InstallActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_install)

        val v1 = intent.data.getQueryParameter("k1")
        val v2 = intent.data.getQueryParameter("k2")

        txtReferrer.text = "V1 : $v1, V2 : $v2"
    }
}
